### PR TITLE
Improve implementation of slider repeat points

### PR DESCRIPTION
**Pull request by Shawdooow**

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 1]
```diff
@@ -131,5 +133,34 @@ public IEnumerable<SliderTick> Ticks
...
+        {
+            get
+            {
+                var length = Curve.Distance;
+                var bouncerDistance = Math.Min(BouncerDistance, length);
+                var repeatDuration = length / Velocity;
+
+                for (var repeat = 0; repeat < RepeatCount; repeat++)
+                {
+                    if (repeat > 0)
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs]: Invert this if statement

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 2]
```diff
@@ -131,5 +133,34 @@ public IEnumerable<SliderTick> Ticks
                 }
             }
         }
+        public IEnumerable<SliderBouncer> Bouncers
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs]: Let's not name them "Bouncers" - "RepeatPoints" sounds better imo.

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 3]
```diff
@@ -73,6 +74,7 @@ public override void ApplyDefaults(ControlPointInfo controlPointInfo, BeatmapDif
 
             Velocity = scoringDistance / timingPoint.BeatLength;
             TickDistance = scoringDistance / difficulty.SliderTickRate;
+            BouncerDistance = Distance;
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs]: Is this needed? Since it's only used from within `Slider`, can we just use `Distance` here?

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game.Rulesets.Osu/Objects/Slider.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Objects/Slider.cs]: Actually might as well just start iterating at repeat = 1

Comment (peppy) [reply osu.Game.Rulesets.Osu/Objects/Slider.cs]: @Shawdooow did you miss this?

File: osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs [hunk 4]
```diff
@@ -41,6 +41,10 @@ protected override void SimulateAutoplay(Beatmap<OsuHitObject> beatmap)
                     // Ticks
                     foreach (var unused in slider.Ticks)
                         AddJudgement(new OsuJudgement { Result = HitResult.Great });
+
+                    //Repeats
+                    foreach (var unused in slider.RepeatPoints)
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: not sure if this will be an issue, but these are being added out of order

Comment (Shawdooow) [reply osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: I actively play lazer with this and it seems to work fine, but do you want it above `ticks` anyway?

Comment (peppy) [reply osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: above isn't even correct; they are interspersed. it's fine this way for now though.

**Comment by peppy** [comment]
Sorry for the delay in merging this one. Looks much better than what was there, so let's go ahead and get this in!

**Review by peppy — APPROVED** [review]

**Pull request merged**